scripts/oes-data-migration-scripts/migration_v3.12.x_to_v4.0.x.py

```diff
@@ -248,11 +248,12 @@ def update_verification_gate_url(json_data, pl_key, execution_str):
         application_name = redis_conn.hget(pl_key, 'application')
         pipeline_name = redis_conn.hget(pl_key, 'name')
         service_id = get_service_id(str(application_name.decode("utf-8")), str(pipeline_name.decode("utf-8")))
-        canary_report_url = json_data['canaryReportURL']
-        json_data['canaryReportURL'] = canary_report_url + '/fromPlugin/' + str(service_id)
-        dump = json.dumps(json_data)
-        redis_conn.hset(pl_key, "stage." + execution_str + ".outputs", dump)
-        logging.info(f"The output after updating verification gate url is: {json_data}")
+        if service_id is not None:
+           canary_report_url = json_data['canaryReportURL']
+           json_data['canaryReportURL'] = canary_report_url + '/fromPlugin/' + str(service_id)
```

Comment:
before appending '/fromPlugin/' to the url we must ensure it is not already present in the url otherwise we will end up corrupting the url. Something similar to what we already have in "update_approval_gate_url" method -
        if navigational_url.find('fromPlugin?instanceId') < 0:

Reply:
Added this condition